azure/Guidelines.md

```diff
@@ -747,6 +749,20 @@ While removing a value from an enum is a breaking change, adding value to an enu
 
 > :ballot_box_with_check: **You SHOULD** use extensible enums unless you are positive that the symbol set will **NEVER** change over time.
 
+### Deprecating Behavior Notification
+
+When the [API Versioning](#API-Versioning) guidance above cannot be followed and the [Azure Breaking Change Reviewers](mailto:azbreakchangereview@microsoft.com) approve a breaking change to the operation/service, the operation/service that is being deprecated must add the `azure-deprecating` response header with a semicolon-delimited string notifying the caller what is being deprecated, when it will no longer function, and a URL linking to more information. The purpose is to inform customers (when debugging/logging responses) that they must take action to modify their call to the service's operation or their call will soon stop working entirely. It is not expected that client code will examine/parse this header's value in any way; it is purely informational to a human being.
+
+:white_check_mark: **DO** add the 'azure-deprercating' header with a string value to all service operations in your service's contract file (cadl/swagger).
+
+:white_check_mark: **DO** include this header in the operation's response _only if_ the operation will stop working in the future and the client _must take_ action in order for it to keep working. NOTE: We do not want to scare customers with this header.
+
+:white_check_mark: **DO** make the header's value a semicolon-delimited string indicating a set of deprecations where each one indicates what is deprecating, when it is deprecating, and a URL to more information. For example: 
+
+```text
+azure-deprecating: api-version=2009-27-07 will stop working on 2022-12-01 (https://azure.microsoft.com/en-us/updates/video-analyzer-retirement);TLS 1.0 & 1.1 will stop working on 2020-10-30 (https://azure.microsoft.com/en-us/updates/azure-active-directory-registration-service-is-ending-support-for-tls-10-and-11/)
```

Comment:
This shows a bit of my doubt with the single header as we don't enforce structure which will be a pain for customers given every service will use its own flavor. I've seen enough occasions of this while maintaining [Azure Deprecation Notices](https://azure-deprecation-notices.cloud/) where some teams do them properly, others not so much.

I believe we should be more explicit in what the required format is and required information that must be provided is.

Another thing that comes to mind - Should they open a review with us to get approval on the header content? Given they will have to update their specs we should also enforce them to provide the samples they want to emit.

Reply:
Also this example is a bit misleading - Is the intent of this header to announce deprecations for the whole API version or just one or more operations of that API version?

Reply:
We always deprecate an entire service version; never some of its operations. A new service version would omit the operations no longer supported. 

Also, it is unusual for an operation to become obsolete entirely. Instead, the operation remains but it breaks in some fashion: different request/response models, different behavior, different return codes, sync into async, non-pageable list becomes a pageable list, etc.

Reply:
Also, I get what you're saying about the structure but there may be some deprecations that don't quite fit the structure and since these strings are for human consumption only, the lack of structure may discourage someone from trying to parse them (which is goodness) and the service team can modify the string in a non-breaking way since there is no contract about their format.